tests/filtered_relation/tests.py

```diff
@@ -279,28 +279,149 @@ def test_as_subquery(self):
         qs = Author.objects.filter(id__in=inner_qs)
         self.assertSequenceEqual(qs, [self.author1])
 
-    def test_with_foreign_key_error(self):
+    def test_with_nested_foreign_key(self):
+        qs = Author.objects.annotate(
+            book_editor_worked_with=FilteredRelation('book__editor', condition=Q(book__title__icontains='book')),
+        ).filter(
+            book_editor_worked_with__isnull=False
+        ).select_related(
+            'book_editor_worked_with'
+        ).order_by(
+            'pk', 'book_editor_worked_with__pk'
+        ).distinct()
+
+        with self.assertNumQueries(1):
+            self.assertQuerysetEqual(qs, [
+                (self.author1, self.editor_a),
+                (self.author2, self.editor_b),
+            ], lambda x: (x, x.book_editor_worked_with))
+
+    def test_with_nested_field(self):
+        qs = Author.objects.annotate(
+            book_editor_worked_with=FilteredRelation('book__editor', condition=Q(book__title__icontains='book')),
```

Comment:
I'm not sure if that is a correct example and sth that we would like to support :thinking:, because conditions across different relations introduce duplicate rows. IMO we should restrict conditions only to starting with `filtered_relation`.

Reply:
What about restricting to those within the `filtered_relation` path? `book__title__icontains` comes from the `book` table, which is accessed on the way to `book_editor`

For more context, I'm using this to do a join on the right side of a left join to convert currencies. I could get more into my specific models if that'd be helpful, but don't want them to distract from the basic functionality. I didn't want to add to the models under test here to replicate what I'm doing, but maybe I should to make it more explicit.

Reply:
@felixxm , I added restrictions around this so the `filtered_relation`'s condition must be within the path to the `filtered_relation`